tutorial/README.md

```diff
@@ -0,0 +1,824 @@
+NAME                        READY   STATUS    RESTARTS   AGE
+data-api-6b489fb85f-v8l2j   1/1     Running   6          8m7s
+db-57655f887d-rwvnm         1/1     Running   0          8m9s
+flights-f5bcb9f69-sfzrg     1/1     Running   0          8m5s
+quakes-77fd856676-wmkhh     1/1     Running   0          8m13s
+ui-6cddfb8bdc-bphhm         1/1     Running   0          8m10s
+weather-6597bdd47f-d8p6h    1/1     Running   0          8m15s
+```
+
+You may see the `data-api` pod in a crash loop. This can happen if the database hasn't completed initialization yet. It may need a few minutes to stabilize.
+
+If you get this far then the application has been successfully deployed. You can use the following command from your control plane VM to remove the application.
+
+```sh
+# you should be logged in as 'azureuser' in the control plane VM
+whoami
+
+# substitute the path to where your files are in this comment
+azoam ./flight-tracker/* | kubectl delete -f -
+```
+
+## Troubleshooting
+
+Remember that you can find the completed sample in `./flight-tracker-solution` for reference if you want to check your answers.
+
+The best way see errors is using the control plane VM. Usually if you make a mistake (omitting a required property, or setting an invalid value) then the application will fail to deploy on the AKS cluster.
+
+You can see this using:
+
+```sh
+kubectl describe applicationconfiguration
+```
+
+You'll see lots of output - with the final lines looking like:
+
+```txt
+Status:
+  Conditions:
+    Last Transition Time:  2020-06-27T19:16:55Z
+    Message:               cannot apply components: cannot apply workload "": cannot get object: resource name may not be empty
+    Reason:                Encountered an error during resource reconciliation
+    Status:                False
+    Type:                  Synced
+Events:
+  Type     Reason                  Age               From                                       Message
+  ----     ------                  ----              ----                                       -------
+  Warning  CannotRenderComponents  21s               oam/applicationconfiguration.core.oam.dev  cannot get component "hello-world": Component.core.oam.dev "hello-world" not found
+  Normal   RenderedComponents      3s (x3 over 21s)  oam/applicationconfiguration.core.oam.dev  Successfully rendered components
+  Warning  CannotApplyComponents   3s (x3 over 21s)  oam/applicationconfiguration.core.oam.dev  cannot apply workload "": cannot get object: resource name may not be empty
+```
+
+In this case the error is: 
+```txt
+cannot apply workload "": cannot get object: resource name may not be empty
+```
+
+---
+
+If you need to fix a mistake the best way to do it is to clear out all of the resources and then apply them again with the correct versions.
+
+```sh
+# using the control plane VM
+azoam flight-tracker/* | kubectl delete -f -
+
+# after fixing the mistake
+azoam flight-tracker/* | kubectl apply -f -
+
+```
+
```

Comment:
Add instructions to delete the resource group as well (just like in the setup) - link there maybe.

Reply:
will do.